### PR TITLE
Adds API consumer docs (closes #6).

### LINKED ISSUE #6: Write API consumer docs.

### PR DESCRIPTION
**Pull request by chuckharmston**
Preview: https://github.com/chuckharmston/universal-search-recommendation/blob/6-api-docs/docs/api.md

### CODE REVIEW

File: docs/api.md [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+
+#### `wikipedia`
+
+If the `result` is a Wikipedia article, the `wikipedia` classifier is applied.
+
+```json
+{
+  "enhancers": {
+    "embedly": {
+      "wikipedia": {
```
Comment (6a68) [inline docs/api.md]: Should this be
```
enhancers: {
  wikipedia: {
```

Comment (chuckharmston) [inline docs/api.md]: Yup, copypasta. Nice catch!

File: docs/api.md [hunk 2]
```diff
@@ -0,0 +1,158 @@
+# API
+
+The recommendation server provides a single API endpoint that, givene a partial search string, will attempt to complete that search and make a destination recommendation.
```
Comment (6a68) [inline docs/api.md]: `given`, not `givene`

File: docs/api.md [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+    "abstract": "Directed by Ridley Scott. With Matt Damon, Jessica Chastain, Kristen Wiig, Kate Mara. During a manned mission to Mars, Astronaut Mark Watney is presumed dead after a...",
+    "title": "The Martian (2015) - IMDb",
+    "url": "http://www.imdb.com/title/tt3659388/"
+  }
+}
+```
+
+There will always be three keys in the root object:
+
+* `enhancers` will be an object containing [context-specific metadata](#enhancers) for the destination recommendation provided in `result`.
```
Comment (6a68) [inline docs/api.md]: Maybe mention that `enhancers` is always present but may be empty if no enhancers were applied

File: docs/api.md [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+* `query` will be an object containing information about the text of the query itself itself:
```
Comment (6a68) [inline docs/api.md]: `itself itself`

File: docs/api.md [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+    * `original` will be a string containing the original query, as passed in the request.
+    * `completed` will be a string containing the query, as completed.
+* `result` includes basic information about a destination recommendation for the completed result:
+    * `abstract` will be a string containing a short textual overview of the result.
+    * `title` will be a string containing a
```
Comment (6a68) [inline docs/api.md]: this line's incomplete

File: docs/api.md [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+
+
+### Enhancers
+
+The server will identify the recommendation as belonging to a zero or more classes of results. For each class it belongs to, additional metadata will be included in the `enhancers` object of the response.
```
Comment (6a68) [inline docs/api.md]: Maybe something about how *special* classes of results can be *enhanced* with data from additional APIs? idk

File: docs/api.md [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+}
+```
+
+There will always be three keys in the root object:
+
+* `enhancers` will be an object containing [context-specific metadata](#enhancers) for the destination recommendation provided in `result`.
+* `query` will be an object containing information about the text of the query itself itself:
+    * `original` will be a string containing the original query, as passed in the request.
+    * `completed` will be a string containing the query, as completed.
+* `result` includes basic information about a destination recommendation for the completed result:
```
Comment (6a68) [inline docs/api.md]: I guess "will include" to keep the tense the same throughout

**Comment by 6a68** [comment]
@chuckharmston Great job on the docs! I can't find any more nits to pick, merging

**Pull request merged**